File: 6.Algorithms: Recursion/3.reverse.py
# Implement a function that reverses a string using iteration...and then recursion!
def reverse_string_iter(str):
  rev = ''
  for i in range(len(str)-1, -1, -1):
    rev += str[i]
  return rev


def reverse_string_recur(str):
  if len(str) < 2:
    return str
  return reverse_string_recur(str[1:]) + str[0]

print(reverse_string_iter('yoyo mastery'))
print(reverse_string_recur('yoyo mastery'))
# should return: 'yretsam oyoy'

File: 6.Algorithms: Recursion/3.reverse.js
//Implement a function that reverses a string using iteration...and then recursion!
function reverseStringIter(str) {
    let rev = '';
    for(let i = str.length - 1; i >= 0; i--){
      rev += str[i];
    }
    return rev;
  }
  
  function reverseStringRecur(str) {
    if(str.length === 1){
      return str;
    }
    return reverseStringRecur(str.slice(1)) + str[0];
  }
  
  console.log(reverseStringIter('yoyo mastery'))
  console.log(reverseStringRecur('yoyo mastery'))
  //should return: 'yretsam oyoy'